### DISCUSSION TITLE
help request: which APISIX version I should use?

### DISCUSSION THREAD

**Discussion by ederparaiso** [apache/apisix#8194] [Q&A] [answered]
I'm a bit confused about APISIX versions. According to oficial docs, 2.15 is the LTS version and 3.0 is beta version, but when I go to installation docs and docker hub they user a 2.99.0 version.
Also, this version doesn't have alpine variation, but its the most recent updated.
Which APISIX version I should use?

**Comment by tokers** [comment, marked as answer]
Indeed `2.99.0` is the `3.0.0-beta`.

**Reply by ederparaiso** [reply]
Whoa, it's a little weird and confusing :thinking: . But thanks for the answer!

ps: Do you know if alpine distribution was deprecated in version 3.0? In docker hub there is no alpine variant for version 2.99 (3.0?)

**Reply by tokers** [reply]
I think maybe @tzssangglass knows the reason.

**Reply by tzssangglass** [reply]
> Do you know if alpine distribution was deprecated in version 3.0? In docker hub there is no alpine variant for version 2.99 (3.0?)

alpine image is abandoned, now use debian

**Reply by ederparaiso** [reply]
Some reason to abandon alpine distro?

**Reply by tzssangglass** [reply]
1. none of the major developers use Alpine as a development environment, nor does CI use Alpine. even if we did, many of the components we rely on would not do so. This means that using Alpine is pretty much a test on production. In fact, we did encounter quite a few problems that only occurred with Alpine.

2. Alpine's clib uses musl. Many components have not released the installation package for musl, so we need to spend a lot of effort to maintain the musl version of APISIX, such as compiling wasmtime from source code and installing Go from apk. We tried Alpine-glibc, but this one has some weird issues, and it's not officially maintained by Alpine.

**Reply by ederparaiso** [reply]
fully understandable. Thanks!
